src/js/_enqueues/wp/webp-fallback.js

```diff
@@ -0,0 +1,130 @@
+
+					var src = images[ j ].src;
+
+					// If there are sources but no sizes, then attempt to replace src through sources. In that case, there is nothing more to replace.
+					if ( media_sources && ! sizes_keys.length ) {
+						// Only modify src if available and the relevant sources are set.
+						if ( src && media_sources['image/webp'] && media_sources['image/jpeg'] ) {
+							src = src.replace( media_sources['image/webp'].file, media_sources['image/jpeg'].file );
+							images[ j ].setAttribute( 'src', src );
+						}
+						continue;
+					}
+
+					var srcset = images[ j ].getAttribute( 'srcset' );
+
+					for ( var k = 0; k < sizes_keys.length; k++ ) {
+						var media_sizes_sources = sizes[ sizes_keys[ k ] ].sources;
+						if ( ! media_sizes_sources || ! media_sizes_sources['image/webp'] || ! media_sizes_sources['image/jpeg'] ) {
+							continue;
+						}
+
+						// Check to see if the image src has any size set, then update it.
+						if ( media_sizes_sources['image/webp'].source_url === src ) {
+							src = media_sizes_sources['image/jpeg'].source_url;
+
+							// If there is no srcset and the src has been replaced, there is nothing more to replace.
+							if ( ! srcset ) {
+								break;
+							}
+						}
+
+						if ( srcset ) {
+							srcset = srcset.replace( media_sizes_sources['image/webp'].source_url, media_sizes_sources['image/jpeg'].source_url );
+						}
+					}
+
+					if ( srcset ) {
+						images[ j ].setAttribute( 'srcset', srcset );
+					}
+
+					if ( src ) {
+						images[ j ].setAttribute( 'src', src );
+					}
+				}
+			} catch ( e ) {
+			}
+		}
+	};
+
+	var restApi = document.getElementById( 'wpFallbackWebpImages' ).getAttribute( 'data-rest-api' );
+
+	var loadMediaDetails = function( nodes ) {
+		var ids = [];
+		for ( var i = 0; i < nodes.length; i++ ) {
+			var node = nodes[ i ];
+			var srcset = node.getAttribute( 'srcset' ) || '';
+
+			if (
+				node.nodeName !== 'IMG' ||
+				( ! node.src.match( /\.webp$/i ) && ! srcset.match( /\.webp\s+/ ) )
+			) {
+				continue;
+			}
+
+			var attachment = node.className.match( /wp-image-(\d+)/i );
+			if ( attachment && attachment[1] && ids.indexOf( attachment[1] ) === -1 ) {
+				ids.push( attachment[1] );
+			}
+		}
+
+		for ( var page = 0, pages = Math.ceil( ids.length / 100 ); page < pages; page++ ) {
+			var pageIds = [];
+			for ( var j = 0; j < 100 && j + page * 100 < ids.length; j++ ) {
+				pageIds.push( ids[ j + page * 100 ] );
+			}
+
+			var jsonp    = document.createElement( 'script' ),
+				restPath = 'wp/v2/media/?_fields=id,media_details&_jsonp=media.wpFallbackWebpImages&per_page=100&include=' + pageIds.join( ',' );
+
+			if ( -1 !== restApi.indexOf( '?' ) ) {
+				restPath = restPath.replace( '?', '&' );
+			}
+
+			jsonp.src = restApi + restPath;
+			document.body.appendChild( jsonp );
+		}
+	};
+
+	try {
+		// Loop through already available images.
+		loadMediaDetails( document.querySelectorAll( 'img' ) );
```

Comment:
I am skeptical about using `document.querySelectorAll( 'img' )` because what if page has lots of images and it will freeze the browser at some point. Do we want to consider view points of browser and then select images that is available from those view points. And then with scrolling, we can catch other images when they are in view port.

Reply:
Good point about potential edge cases where this lookup would fail.

> Do we want to consider view points of browser and then select images that is available from those view points. And then with scrolling, we can catch other images when they are in view port.

This feels overly complex for the goal of providing backwards compatibility for a few remaining browser users.  How about setting a sensible limit to the number of images we would swap. eg. limit the total number of images handled to 100?